tripal/src/Form/TripalContentFieldsForm.php

```diff
@@ -0,0 +1,201 @@
+      '#empty' => t('No new fields were discovered that could be added.'),
+    ];
+
+    // Add elements for the existing fields tab.
+    $existing_fields = [];
+    $existing_defaults = [];
+    foreach ($fields['existing'] as $field) {
+      $existing_fields[$field['name']] = $field['label'] . " (" . $field['name'] . "): " . $field['description'];
+      $existing_defaults[] = $field['name'];
+    }
+    $existing_fields_desc = t('The following is a list of fields that were discovered but which are already attached to this content type. They are shown here but cannot be added again.');
+    if (empty($existing_fields)) {
+      $existing_fields_desc = t('No new fields were found for this content type');
+    }
+    $form['existing_fields_details']['existing_fields_list'] = [
+      '#type' => 'checkboxes',
+      '#title' => 'Existing Fields',
+      '#description' => $existing_fields_desc,
+      '#description_display' => 'before',
+      '#options' => $existing_fields,
+      '#default_value' => $existing_defaults,
+      '#disabled' => TRUE,
+    ];
+
+    // Add elements for the invalid fields tab.
+    $invalid_fields = [];
+    $invalid_defaults = [];
+    foreach ($fields['invalid'] as $field) {
+      $invalid_fields[$field['name']] = $field['label'] . " (" . $field['name'] . "): " . $field['description']. '. Invalid Reason: ' . $field['invalid_reason'];;
+    }
+    $invalid_fields_desc = t('The following fields do not pass validation tests. They need correction by the module developer and cannot be added.');
+    if (empty($invalid_fields_desc)) {
+      $invalid_fields_desc = t('All fields passed validation tests!');
+    }
+    $form['invalid_fields_details']['invalid_fields_list'] = [
+      '#type' => 'checkboxes',
+      '#title' => 'Invalid Fields',
+      '#description' => $invalid_fields_desc,
+      '#description_display' => 'before',
+      '#options' => $invalid_fields,
+      '#default_value' => $invalid_defaults,
+      '#disabled' => TRUE,
+    ];
+
+    $submit_disabled = FALSE;
+    if (empty($new_fields)) {
+      $submit_disabled = TRUE;
+    }
+    $form['sumbmit'] = [
+      '#type' => 'submit',
+      '#value' => 'Add Fields',
+      '#disabled' => $submit_disabled,
+    ];
+
+    return $form;
+  }
+
+  /**
+   *
+   * {@inheritDoc}
+   * @see \Drupal\Core\Form\FormInterface::submitForm()
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+
+    $messenger = \Drupal::messenger();
+
+    // Get submitted form elements.
+    $tripal_entity_type = $form_state->getValue('tripal_entity_type');
+    $fields = $form_state->getValue('discovered_fields');
+    $add_fields = $form_state->getValue('new_fields_choice');
+
+    /** @var \Drupal\tripal\Services\TripalFieldCollection $tripal_fields **/
+    $tripal_field_collection = \Drupal::service('tripal.tripalfield_collection');
+    $fields = $tripal_field_collection->discover($tripal_entity_type);
```

Comment:
This is the second time you defined the `$fields` variable in this function. I'm assuming we can remove the one on line 167? In which case we can also likely remove that form element from the build() as well.